### DISCUSSION TITLE
Are you supposed to see request in network tab if fetching data on server?

### DISCUSSION THREAD

**Discussion by Rc85** [TanStack/query#6872] [Q&A]
I was under the impression that if I do the fetching on a server component, I will not see the request in the network's tab and I should see the data right away upon completing page load?

I am following the app router's tutorial [here](https://tanstack.com/query/latest/docs/framework/react/guides/advanced-ssr#prefetching-and-dehydrating-data), but I'm still seeing the request in the network's tab and moreover, the loading indicator from the `isLoading` state.

Can someone enlighten me?

**Comment by TkDodo** [comment]
show a codesandbox reproduction please. Likely you're seeing a _refetch_ from the client.

**Reply by Rc85** [reply]
Here is a [codesandbox](https://codesandbox.io/p/devbox/silent-darkness-7nflt4?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clska6oxu0007356mfe8oyj7q%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clska6oxu0002356mp5f4fqqp%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clska6oxu0004356myosiy50v%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clska6oxu0006356mj3i7dk2v%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clska6oxu0002356mp5f4fqqp%2522%253A%257B%2522id%2522%253A%2522clska6oxu0002356mp5f4fqqp%2522%252C%2522tabs%2522%253A%255B%255D%257D%252C%2522clska6oxu0006356mj3i7dk2v%2522%253A%257B%2522id%2522%253A%2522clska6oxu0006356mj3i7dk2v%2522%252C%2522activeTabId%2522%253A%2522clskcbk5f001a356m8712l0oe%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522TASK_PORT%2522%252C%2522port%2522%253A3000%252C%2522taskId%2522%253A%2522dev%2522%252C%2522id%2522%253A%2522clskcbk5f001a356m8712l0oe%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%252F%2522%257D%255D%257D%252C%2522clska6oxu0004356myosiy50v%2522%253A%257B%2522id%2522%253A%2522clska6oxu0004356myosiy50v%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clska6oxu0003356myd40kl9t%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clska6oxu0003356myd40kl9t%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D) but I can't get set it up for some reason. The `HydrationBoundary` is throwing an error saying `Error: Only plain objects, and a few built-ins, can be passed to Client Components from Server Components. Classes or null prototypes are not supported.`

If you can help, that would be nice. If not, I'll resort to using fetch on the server.

**Reply by TkDodo** [reply]
```
    queryFn: () =>
      fetch("https://mocki.io/v1/d4867d8b-b5d5-4a48-a4ab-79131b5809b8"),
```

you're returning the whole fetch response, not just the `.json()` part. This isn't serializable

**Reply by Rc85** [reply]
I'm an idiot lol. Fixed it but now I'm getting Maximum call stack size exceeded and it seems to be from the `HydrationBoundary` because if I remove it, it renders.

**Reply by TkDodo** [reply]
I mean, are you trolling me? Now you return a full axios response and not just the `.data` part, and that is again a recursive, non-serializable thing 🙄

**Reply by Rc85** [reply]
That's strange because that's how I've been doing it in my personal projects and it works. Anyways, I fixed it now and as you can see in the network tab, the request to mock.io is there. Can you confirm I set up the prefetch and useQuery correctly?

With all due respect, I was not trolling you.

**Reply by TkDodo** [reply]
> Likely you're seeing a refetch from the client.

this is still the case. With the default staleTime of zero, data gets into the cache from the server, but then you get a refetch on the client.

this is well documented: https://tanstack.com/query/v5/docs/framework/react/guides/ssr#full-nextjs-pages-router-example

```
  const [queryClient] = React.useState(
    () =>
      new QueryClient({
        defaultOptions: {
          queries: {
            // With SSR, we usually want to set some default staleTime
            // above 0 to avoid refetching immediately on the client
            staleTime: 60 * 1000,
          },
        },
      }),
  )
```
